src/codemodder/result.py

```diff
@@ -68,18 +68,39 @@ def match_location(self, pos: CodeRange, node: cst.CSTNode) -> bool:
 
 
 @dataclass(kw_only=True)
-class SarifResult(Result, ABCDataclass):
+class SASTResult(Result):
+    finding_id: str
+
+
+@dataclass(kw_only=True)
+class SarifResult(SASTResult, ABCDataclass):
     location_type: ClassVar[Type[SarifLocation]]
 
     @classmethod
     def from_sarif(
         cls, sarif_result, sarif_run, truncate_rule_id: bool = False
     ) -> Self:
+        # avoid circular import
+        from core_codemods.semgrep.api import semgrep_url_from_id
+
         return cls(
-            rule_id=cls.extract_rule_id(sarif_result, sarif_run, truncate_rule_id),
+            rule_id=(
+                rule_id := cls.extract_rule_id(
+                    sarif_result, sarif_run, truncate_rule_id
+                )
+            ),
             locations=cls.extract_locations(sarif_result),
             codeflows=cls.extract_code_flows(sarif_result),
             related_locations=cls.extract_related_locations(sarif_result),
+            finding_id=rule_id,
+            finding=Finding(
+                id=rule_id,
+                rule=Rule(
+                    id=rule_id,
+                    name=rule_id,
+                    url=semgrep_url_from_id(rule_id),
```

Comment:
This is specific to Semgrep but it looks like this is being added to a more general `SarifResult` class?

Reply:
yeah do you think this is wrong? while it's a generic sarif result, wouldn't all sarif results have to have findings too?

Reply:
My concern is specifically about the call to `semgrep_url_from_id`, which only applies to Semgrep results, whereas this class appears to be used for multiple SARIF-based tools.

Reply:
wooops Im an idiot, of course! Good review Dan, will update